OLS-1356: Document Swagger support for OLS (#109931)
openshift/openshift-docs: modules/ols-con-openshift-lightspeed-rest-api-specifications.adoc

// Module included in the following assemblies:
// lightspeed-docs-main/operate/ols-using-openshift-lightspeed.adoc

:_mod-docs-content-type: CONCEPT
[id="ols-con-openshift-lightspeed-rest-api-specifications_{context}"]
= {ols-long} REST API specifications

[role="_abstract"]

> {ols-long} REST API specifications use OpenAPI 3.1 to provide interactive documentation and machine-readable files. These automatically generated specifications help you explore API endpoints and integrate the service into your environment.

Reviewer: ```suggestion
Use the OpenAPI 3.1 specifications for {ols-long} to access interactive REST API documentation and download machine-readable files for environment integration.
```

The {ols-long} REST API utilizes OpenAPI 3.1 specifications. The system automatically generates the specification from the route definitions, and you do not need to manually configure separate specification files. 

[WARNING]
====

> Do not expose the {ols-long} service directly to the public internet without enabling authentication. In a production {ocp-product-title} environment, ensure the service is fronted by `kube-rbac-proxy` for authentication. 

Reviewer: ```suggestion
Do not expose the {ols-long} service directly to the public internet without enabling authentication. In a production {ocp-product-title} environment, ensure that you use `kube-rbac-proxy` to front the service for authentication. 
```
